js/main.js

```diff
@@ -425,6 +427,26 @@ function setCity(cityID, createNewHistoryEntry) {
     });
 }
 
+/*
+ * Checks the cityID for unexpected characters or if it is too long.
+ *
+ * In case the cityID is invalid, it emits a console.error message.
+ *
+ * Returns the cityID if the cityID is valid, otherwise the DEFAULT_CITY_ID.
+ */
+function sanitizeCityID(cityID) {
+  // the regexp also contains German Umlaute äöü
+  // from here https://stackoverflow.com/a/22017800 by
+  // IProblemFactory (https://stackoverflow.com/users/140070/iproblemfactory)
+  var validCityRegexp = '^[\-a-z0-9_\u00e4\u00f6\u00fc]{1,40}$';
+  var regexp = new RegExp(validCityRegexp, 'i');
+  if (!regexp.test(cityID)) {
+    console.error('Invalid city id hash. Loading  "' + DEFAULT_CITY_ID +
```

Comment:
The error should output the invalid `cityID` so it is easier to diagnose the error.